github-fetcher/json-orm.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/go-github/github"
+)
+
+// Issue is a pull-request or issue. It's format fits into the ORM
+type Issue struct {
+	ID             int
+	Labels         []Label
+	Title          string
+	Body           string
+	User           string
+	Assignee       *string
+	State          string
+	Comments       int
+	IsPR           bool
+	IssueClosedAt  *time.Time
+	IssueCreatedAt time.Time
+	IssueUpdatedAt time.Time
+}
+
+// NewIssue creates a new (orm) Issue from a github Issue
+func NewIssue(gIssue *github.Issue) *Issue {
+	if gIssue.Number == nil ||
+		gIssue.Title == nil ||
+		gIssue.User == nil ||
+		gIssue.User.Login == nil ||
+		gIssue.State == nil ||
+		gIssue.Comments == nil ||
+		gIssue.CreatedAt == nil ||
+		gIssue.UpdatedAt == nil {
+		glog.Errorf("Issue is missing mandatory field: %+v", gIssue)
```

Comment:
Agreed, use multiple returns here and let the caller decide what to do with the error.